tests/test_models/test_losses/test_focal_loss.py

```diff
@@ -0,0 +1,1618 @@
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        2.1043241024017334,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        0.9553812742233276,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+        3.6949386596679688,
+    ]
+    weights = [
+        torch.tensor([[0.4118, 0.0461, 0.9419], [0.4523, 0.8014, 0.4209],
+                      [0.3738, 0.7583, 0.2685], [0.4707, 0.9720, 0.3626],
+                      [0.2771, 0.2611, 0.4687], [0.2841, 0.7127, 0.7278],
+                      [0.4920, 0.2341, 0.7262], [0.2499, 0.4702, 0.9717]]),
+        torch.tensor(
+            [0.4344, 0.8858, 0.7353, 0.5068, 0.5316, 0.1429, 0.5781, 0.7336]),
+        None
+    ]
+    class_weights = [[
+        0.052917540073394775, 0.27414458990097046, 0.5211336612701416
+    ], None]
+    ignore_indexes = [255, 4]
+    index = 0
+    for reduce in ['none', 'mean', 'sum']:
+        for weight in weights:
+            for class_weight in class_weights:
+                for target in [fake_target, fake_target_one_hot]:
+                    for ignore_index in ignore_indexes:
+                        for alpha in [[0.4] * 3, 0.4, [0.4] * 3, 0.4]:
+                            gpu_result = gpu_results[index]
+                            if isinstance(gpu_result, float):
+                                gpu_result = torch.tensor(gpu_result)
+                            index += 1
+                            focal_loss.class_weight = class_weight
+                            focal_loss.alpha = alpha
+                            cpu_result = focal_loss(
+                                fake_pred,
+                                target,
+                                weight=weight,
+                                ignore_index=ignore_index,
+                                reduction_override=reduce)
+                            assert torch.allclose(cpu_result, gpu_result)
```

Comment:
The nest is too deep.
A single test for each case (covering all the branches) is sufficient.

Reply:
ok